releasenotes/upgrade.rst

```diff
@@ -122,4 +122,61 @@ So start with upgrading the references right away.
 
 Especially for Facades to relational databases: match the version of EntityFrameworkCore that the new version of Firely Server is using. Check the list of changes to see whether we upgraded.
 
-.. _SDK release notes: https://github.com/FirelyTeam/firely-net-sdk/releases
\ No newline at end of file
+.. _SDK release notes: https://github.com/FirelyTeam/firely-net-sdk/releases
+
+.. _migrations:
+
+Migrations
+----------
+
+Sometimes, an upgrade of the database schema is needed in order to introduce performance improvements or enable new features. Depending on the database that is used, Firely Server provides different techniques to migrate the data. Most migrations can be performed automatically whereas in some cases, manual migrations are preferable.
+
+One of these cases is a scenario in which Firely Server operates at scale, handling billions of resources. In such a scenario, manual migrations are preferable as they might take longer than expected and come with challenges. In this section, we will elaborate on some of these challenges and how to mitigate them. 
+
+MongoDB
+^^^^^^^
+
+For Firely Server instances running on MongoDB, we consider databases with more than 500GB as large. Migrations for these databases take a long time which is why we introduced external migration scripts with Firely Server 4.5.0. These external scripts are written in JavaScript and allow to exert more control at the cost of some convenience. 
+
+The migration scripts usually perform two operations in this order: 
+   #. Migrate the existing data using the ``updateMany()`` operator.
+   #. Update the system information document to the next version.
+
+While migrating large databases, it is possible that the migration script times out, is interrupted or encounters other unforeseen circumstances. In this case, migrations initiated by the ``updateMany()`` operator will continue running in the background. The update of the schema version, however, will not succeed and needs to be performed manually. Follow these steps to do so: 
+
+#. Connect to your MongoDB host.
+#. Open a MongoDB shell by executing ``mongo`` from the command line.
+#. Switch to the vonk database: ``use vonkdata``
+#. Check whether the current migration is still running with: ``db.currentOp()`` 
```

Comment:
How do you recognize the migration operation? Is that obvious?

Reply:
Kind of, but I improved the wording anyways!